addon-test-support/@ember/test-helpers/test-metadata.ts

```diff
@@ -0,0 +1,52 @@
+import { BaseContext } from './setup-context';
+
+export interface ITestMetadata {
+  testName?: string;
+  setupTypes: string[];
+  usedHelpers: string[];
+  [key: string]: any;
+
+  readonly isRendering: boolean;
+  readonly isApplication: boolean;
+}
+
+export class TestMetadata implements ITestMetadata {
+  [key: string]: any;
+  testName?: string;
+  setupTypes: string[];
+  usedHelpers: string[];
+
+  constructor() {
+    this.setupTypes = [];
+    this.usedHelpers = [];
+  }
+
+  get isRendering() {
+    return (
+      this.setupTypes.length === 2 &&
+      this.setupTypes.includes('setupRenderingContext') &&
```

Comment:
Sadly, we can't rely on `.includes`:

```suggestion
      this.setupTypes.indexOf('setupRenderingContext') > -1 &&
```